clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -123,6 +123,36 @@ def ConstStructAttr : CIR_Attr<"ConstStruct", "const_struct",
   let genVerifyDecl = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// IntegerAttr
+//===----------------------------------------------------------------------===//
+
+def IntAttr : CIR_Attr<"Int", "int", [TypedAttrInterface]> {
+  let summary = "An Attribute containing a integer value";
+  let description = [{
+    An integer attribute is a literal attribute that represents an integral
+    value of the specified integer type.
+  }];
+  let parameters = (ins AttributeSelfTypeParameter<"">:$type, "APInt":$value);
```

Comment:
I see you picked `APInt` as your storage, is that leaner than holding a `mlir::IntegerAttr`? Or is the later already implemented in terms of the former?

Reply:
> MLIR's builtin dialect implements mlir::IntegerAttr by wrapping it around an APInt to store its value

Sorry for the confusion, this is the answer I was looking for, thanks!